src/app/components/annotator/text-annotator/text-annotator.component.ts

```diff
@@ -0,0 +1,63 @@
+import { Component, OnDestroy } from '@angular/core';
+import { Subscription } from 'rxjs';
+import { AnnotatorColors, AppConfig } from 'src/app/app.config';
+import { AnnotatorService } from 'src/app/services/annotator/annotator.service';
+
+interface AnnotatorPosition {
+  x: number;
+  y: number;
+}
+
+interface AnnotationType {
+  type: string | undefined;
+}
+@Component({
+  selector: 'evt-text-annotator',
+  templateUrl: './text-annotator.component.html',
+  styleUrls: ['./text-annotator.component.scss'],
+})
+export class TextAnnotatorComponent implements OnDestroy {
+  private subscriptions: Subscription[] = [];
+  public highlightColors: AnnotatorColors = AppConfig.evtSettings.edition.annotatorColors;
+  public selectedText: string;
+  public showAdder = false;
+  public menuPosition: AnnotatorPosition = { x: 0, y: 0 };
+  public annoType: AnnotationType = {type: undefined};
```

Comment:
if it is a string let it be a string. I was asking if there exist a type like `type AnnotationType = X | Y | Z` to describe better the possible values. Wapping a srtring with an object does not seems useful.